src/hotspot/share/runtime/os.hpp

```diff
@@ -100,19 +100,30 @@ class os: AllStatic {
 #endif
 
  public:
-  enum { page_sizes_max = 9 }; // Size of _page_sizes array (8 plus a sentinel)
+
+  // A simple value class holding a set of page sizes (similar to sigset_t)
+  class PagesizeSet {
+    uintx _v;
+  public:
+    PagesizeSet() : _v(0) {}
+    void add(size_t pagesize);
+    bool is_set(size_t pagesize) const;
```

Comment:
Could this be private. It's an implementation detail that you use a bit set and have "set" bits. If not, maybe consider a more high-level name. is_added, is_registered, contains?

Reply:
I'd prefer if this stays public, I've already found it useful, but I agree that the name could be more informative. Perhaps contains(size_t pagesize)?

Reply:
Cannot be private since its used from other nested classes (os::<platform>). But I can rename it to contains()